src/plugins/table/table.less

```diff
@@ -12,66 +15,76 @@
 .TableDisplay-cell {
   text-align: left;
   font-weight: normal;
-  border: 1px solid @edu-border-color-base;
-  padding: @table-plugin-cell-content-padding;
+  border: 1px solid @table-plugin-cell-border-color;
+  padding: @table-plugin-cell-padding;
 
   &.TableDisplay-cell--header {
-    background-color: @edu-background-color-base;
+    background-color: @table-plugin-cell-header-background-color;
   }
 
   &.TableDisplay-cell--body {
-    background-color: @edu-white;
+    background-color: @table-plugin-cell-body-background-color;
   }
 }
 
 .TableEditor {}
 
 .TableDesigner {}
 
-.TableDesigner-table {
-  .TableDisplay-table;
+.TableDesigner-grid {
+  display: grid;
   margin: -20px 0 0 -20px;
   width: ~"calc(100% + 20px)";
 }
 
-.TableDesigner-tableCell {
-
-  &.TableDesigner-tableCell--content {
-    .TableDisplay-cell;
+.TableDesigner-gridCell {
+  display: grid;
+  align-items: center;
+  justify-items: stretch;
+
+  &.TableDesigner-gridCell--content {
+    text-align: left;
+    font-weight: normal;
+    border-style: solid;
+    border-left-width: 0;
+    border-top-width: 0;
+    border-right-width: 1px;
+    border-bottom-width: 1px;
+    border-color: @table-plugin-cell-border-color;
+    padding: @table-plugin-cell-padding;
     position: relative;
     background-color: @edu-white;
 
-    // Fixes this bug in Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=688556
-    // Background of cell paints over border when borders are collapsed and tbody, row, or cell is relatively positioned
-    background-clip: padding-box;
+    &.TableDesigner-gridCell--firstInRow {
+      border-left-width: 1px;
+    }
+
+    &.TableDesigner-gridCell--firstInColumn {
+      border-top-width: 1px;
+    }
   }
 
-  &.TableDesigner-tableCell--rowHeader {
-    .TableDisplay-cell;
+  &.TableDesigner-gridCell--rowHeader {
+    padding: @table-plugin-cell-padding;
     text-align: center;
-    vertical-align: middle;
-    border: none;
     width: 20px;
   }
 
-  &.TableDesigner-tableCell--columnHeader {
-    .TableDisplay-cell;
+  &.TableDesigner-gridCell--columnHeader {
+    padding: @table-plugin-cell-padding;
     text-align: center;
-    vertical-align: middle;
-    border: none;
     height: 20px;
```

Comment:
Room for some small simplification here with merging the row and column CSS class variations into one, since they don't diverge.

Reply:
Ah ok, I see 😬